pynamodb/attributes.py

```diff
@@ -525,6 +535,36 @@ def deserialize(self, value):
         return json.loads(value)
 
 
+class TTLAttribute(Attribute):
+    """
+    An attribute that signifies when how long this item can live for.
+    This can take in an int, timedelta, or datetime.  If a timedelta is passed in,
+    we assume the expiration time is datetime.utcnow() + timedelta.
+    """
+    attr_type = NUMBER
+
+    def serialize(self, value):
+        """
+        Return the epoch representation (in seconds).
+        """
+        if value is None:
+            return None
+        elif isinstance(value, int):
+            return value
+        elif isinstance(value, timedelta):
+            return int(time.time() + value.total_seconds())
+        elif isinstance(value, datetime):
+            return int((value - datetime(1970,1,1)).total_seconds())
```

Comment:
I think it'll break with tz-aware datetimes:
```
TypeError: can't subtract offset-naive and offset-aware datetimes
```

In Python 3 we have `datetime.timestamp`, but for compatibility I think we'd have to do:
```
import calendar

return calendar.timegm(value.timetuple())
```

From testing:
```py
>>> import calendar 
>>> import time
>>> from datetime import datetime
>>> time.time() - calendar.timegm(datetime.utcnow().timetuple())
0.7275009155273438
>>> time.time() - calendar.timegm(datetime.now(tz=tzutc()).timetuple())
0.18059110641479492
```
(first one is naive, second is tz-aware, in both cases the delta is less than 1 second)

Reply:
Correct.. timestamp() breaks in python2.  That's why I didn't use it.  I'll play around with something..

Reply:
Also I'm fairly certain clendar.timegm only works for.. gm (ie: utc time)

Just verified.. 
```
>>> calendar.timegm(datetime.utcnow().timetuple()) - time.time()
-0.058157920837402344
>>> calendar.timegm(datetime.now().timetuple()) - time.time()
-25200.54909801483
```

Reply:
`datetime.now()` returns a naive timezone in your local timezone, but it's impossible to tell that it's not UTC (i.e. the results of `datetime.now()` and `datetime.utcnow()` are indistinguishable).

Reply:
You're right, though!
```
import calendar
import time
from datetime import datetime
from dateutil import tzutc, tzlocal
```
then
```
>>> time.time() - calendar.timegm(datetime.now(tz=tzlocal()).utctimetuple())
25200.258514881134
```

With `utctimetuple` it seems to be OK, though:
```
>>> time.time() - calendar.timegm(datetime.now(tz=tzlocal()).utctimetuple())
0.7876191139221191
>>> time.time() - calendar.timegm(datetime.now(tz=tzutc()).utctimetuple())
0.14149904251098633
```
```

Reply:
I think this is almost right..

```
>>> calendar.timegm(datetime(2019, 6,5, tzinfo=pytz.timezone('America/New_York')).utctimetuple()) - 1559707200 # 1559707200 is 9PM 6/4/2019 SF time
3600
```